packages/graphql/lib/src/links/websocket_link/websocket_client.dart

```diff
@@ -86,30 +87,43 @@ class SocketClientConfig {
   /// Warning: if you want to listen to the listen to the stream,
   /// wrap your channel with our [GraphQLWebSocketChannel] using the `.forGraphQL()` helper:
   /// ```dart
-  /// connect: (url, protocols) {
+  /// customConnect: (url, protocols) {
   ///    var channel = WebSocketChannel.connect(url, protocols: protocols)
   ///    // without this line, our client won't be able to listen to stream events,
   ///    // because you are already listening.
   ///    channel = channel.forGraphQL();
   ///    channel.stream.listen(myListener)
   ///    return channel;
-  /// }
+  /// }socketChannel
   /// ```
   ///
   /// To supply custom headers to an IO client:
   /// ```dart
   /// connect: (url, protocols) =>
   ///   IOWebSocketChannel.connect(url, protocols: protocols, headers: myCustomHeaders)
   /// ```
-  final WebSocketConnect connect;
-
-  static Future<WebSocketChannel> defaultConnect(
-    Uri uri,
-    Iterable<String>? protocols,
-  ) async {
+  final WebSocketConnect? customConnect;
+
+  /// Custom header to add inside the client
+  final Map<String, dynamic>? customHeaders;
```

Comment:
When is this used? And maybe:

```suggestion
  final Map<String, dynamic>? headers;
```

Reply:
It is used only we are not using flutter when, in particular in this function

```
   return defaultConnectPlatform(
      uri,
      protocols,
      headers: headers,
    );
```

P.S: I Like the `header` name, thanks